pkg/image/api/helper.go

```diff
@@ -843,3 +843,37 @@ func SetContainerImageEntrypointAnnotation(annotations map[string]string, contai
 	s, _ := json.Marshal(cmd)
 	annotations[key] = string(s)
 }
+
+// ImageHasSignature returns true if the given signature is already stored in the image.
+func ImageHasSignature(image *Image, s *ImageSignature) bool {
+	for _, signature := range image.Signatures {
+		if signature.Type == s.Type && signature.Blob == s.Blob {
```

Comment:
If `type` is supposed to be something like `gpg` or `base64`, this is not what we want: we have an array of signatures so that we can attach multiple signature to a single image, but we only have one signature type (and we wouldn’t want to add new types just to support multisignatures).